test/integration/targets/vyos_facts/tests/cli/basic_facts.yaml

```diff
@@ -0,0 +1,36 @@
+- name: get host name
+  vyos_command:
+    commands:
+      - show host name
+  register: vyos_host
+
+- name: get version info
+  vyos_command:
+    commands:
+      - show version
+  register: vyos_version
+
+- name: collect all facts from the device
+  vyos_facts:
+    gather_subset: all
+  register: result
+
+- name: "check that all the fields which are always returned are present"
+  assert:
```

Comment:
while I figure you have just one assert for clean and tidy code, I think it's better if you assert right after every test.
That way we ensure we bail out early in case an assert fails, instead of  running al tasks, register and assert all in block.